### PR TITLE
chore: Bump up to Go 1.19, minimum 1.18

### PR DESCRIPTION
**Pull request by francislavoie**

### CODE REVIEW

**Comment by francislavoie** [comment]
Before merging, we'll need to update the required CI tests (add 1.19, remove 1.17)

**Comment by francislavoie** [comment]
Can we make use of this at all to make `./caddy version` suck less for development `go build` builds? https://tip.golang.org/doc/go1.18#go-version

**Comment by francislavoie** [comment]
We'll probably want to adjust things to use the new IP type https://tip.golang.org/doc/go1.18#netip

**Comment by francislavoie** [comment]
We'll probably want to mention this in the release notes: https://tip.golang.org/doc/go1.18#sha1

**Comment by francislavoie** [comment]
We can probably update a lot of `strings.Index` usages to use the new `strings.Cut` in many cases https://tip.golang.org/doc/go1.18#strings

File: modules/caddyhttp/staticresp.go [hunk 1]
```diff
@@ -283,7 +283,7 @@ func cmdRespond(fl caddycmd.Flags) (int, error) {
 	// build headers map
 	hdr := make(http.Header)
 	for i, h := range respondCmdHeaders {
-		key, val, found := cut(h, ":") // TODO: use strings.Cut() once Go 1.18 is our minimum
+		key, val, found := strings.Cut(h, ":")
```
Comment (mholt) [inline modules/caddyhttp/staticresp.go:286]: Lol great, this TODO only lasted about a day :sweat_smile:

**Review by mholt — APPROVED** [review]
Whew, I reviewed every changed line in this PR!

> Before merging, we'll need to update the required CI tests (add 1.19, remove 1.17)

Done. Well, I removed 1.17, can't add 1.19 until this is merged I think.

> Can we make use of this at all to make ./caddy version suck less for development go build builds? [tip.golang.org/doc/go1.18#go-version](https://tip.golang.org/doc/go1.18#go-version)

Yes, absolutely! Maybe a separate PR? I don't mind working on this one (unless you want to).

> We'll probably want to adjust things to use the new netip.Addr type [tip.golang.org/doc/go1.18#netip](https://tip.golang.org/doc/go1.18#netip)

Yeah, although that might be pretty invasive. Definitely a separate PR, and maybe after our next release (like after 2.6).

> We'll probably want to mention this in the release notes: [tip.golang.org/doc/go1.18#sha1](https://tip.golang.org/doc/go1.18#sha1)

Good idea.

> We can probably update a lot of strings.Index usages to use the new strings.Cut in many cases [tip.golang.org/doc/go1.18#strings](https://tip.golang.org/doc/go1.18#strings)

That's quite likely, yeah.

Thanks for the quick PR!

**Comment by francislavoie** [comment]
Yeah my above notes are mostly TODO that should be made into separate issues probably. I don't think I'll work on those myself right now, just what I noticed from reading the 1.18 and 1.19 releases notes with my Caddy goggles on.

**Review by mohammed90 — APPROVED** [review]
Looking good! 🎉 @mholt has already beat me to the details 😁

**Review by Conan-Kudo — APPROVED** [review]
LGTM, we should be good with RHEL 9.1.

cc: @carlwgeorge

**Pull request merged**